Tribler/Test/Core/TFTP/test_tftp_packet.py

```diff
@@ -0,0 +1,102 @@
+        """
+        Testing whether decoding the options raises InvalidPacketException if no options are found
+        """
+        _decode_options({}, "\0a\0", 0)
+
+    @raises(InvalidPacketException)
+    def test_decode_options_no_value(self):
+        """
+        Testing whether decoding the options raises InvalidPacketException if no value is found
+        """
+        _decode_options({}, "b\0\0", 0)
+
+    @raises(InvalidOptionException)
+    def test_decode_options_unknown(self):
+        """
+        Testing whether decoding the options raises InvalidOptionException if an invalid option is found
+        """
+        _decode_options({}, "b\0a\0", 0)
+
+    @raises(InvalidOptionException)
+    def test_decode_options_invalid(self):
+        """
+        Testing whether decoding the options raises InvalidOptionException if an invalid option is found
+        """
+        _decode_options({}, "blksize\0a\0", 0)
+
+    @raises(InvalidPacketException)
+    def test_decode_data(self):
+        """
+        Testing whether an InvalidPacketException is raised when our incoming data is too small
+        """
+        _decode_data(None, "aa", 42)
+
+    @raises(InvalidPacketException)
+    def test_decode_ack(self):
+        """
+        Testing whether an InvalidPacketException is raised when our incoming ack has an invalid size
+        """
+        _decode_ack(None, "aa", 42)
+
+    @raises(InvalidPacketException)
+    def test_decode_error_too_small(self):
+        """
+        Testing whether an InvalidPacketException is raised when our incoming error has an invalid size
+        """
+        _decode_error(None, "aa", 42)
+
+    @raises(InvalidPacketException)
+    def test_decode_error_no_message(self):
+        """
+        Testing whether an InvalidPacketException is raised when our incoming error has an empty message
+        """
+        _decode_error({}, "aa\0", 0)
+
+    @raises(InvalidPacketException)
+    def test_decode_error_invalid_pkg(self):
+        """
+        Testing whether an InvalidPacketException is raised when our incoming error has an invalid structure
+        """
+        _decode_error({}, "aaa\0\0", 0)
+
+    @raises(InvalidPacketException)
+    def test_decode_packet_too_small(self):
+        """
+        Testing whether an InvalidPacketException is raised when our incoming packet is too small
+        """
+        decode_packet("aaa")
+
+    @raises(InvalidPacketException)
+    def test_decode_packet_opcode(self):
+        """
+        Testing whether an InvalidPacketException is raised when our incoming packet contains an invalid opcode
+        """
+        decode_packet("aaaaaaaaaa")
+
+    def test_encode_packet_error(self):
+        """
+        Testing whether the encoding of an error packet is correct
+        """
+        encoded = encode_packet({'opcode': OPCODE_ERROR, 'session_id': 123, 'error_code': 1, 'error_msg': 'hi'})
+        self.assertEqual(decode_packet(encoded)['opcode'], OPCODE_ERROR)
```

Comment:
The 'positive' code is already covered quite well by a bigger test (`test_remote_torrent_handler.py` iirc) so I didn't spent much time on creating tests for successful runs.